tendermint-rs/src/chain/id.rs

```diff
@@ -21,6 +21,7 @@ pub const MAX_LENGTH: usize = 50;
 pub struct Id([u8; MAX_LENGTH]);
 
 impl Id {
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
Alternatively this could be `impl FromStr for Id`

Reply:
We could still remove ` pub fn new` and place the logic into `from_str`?

Reply:
SGTM